### PR TITLE
Enable building osrm-extract-conditionals on travis

### PR DESCRIPTION
**Pull request by oxidase**
# Issue

PR #3415  did not include building on travis `osrm-extract-conditionals` and was tested only locally.
This PR adds installation of `libshp` and building `osrm-extract-conditionals` on travis.

/cc @karenzshea 

## Tasklist
 - [x] review
 - [x] adjust for comments

## Requirements / Relations
PR #3415 
https://github.com/mapbox/mason/pull/364

### CODE REVIEW

**Comment by karenzshea** [comment]
This is waiting on https://github.com/mapbox/mason/pull/364

**Comment by oxidase** [comment]
`osrm-extract-conditionals` is enabled in mason builds https://travis-ci.org/Project-OSRM/osrm-backend/jobs/207999217#L1527

File: src/contractor/contractor.cpp [hunk 1]
```diff
@@ -117,6 +117,7 @@ template <typename T> inline bool is_aligned(const void *pointer)
     return reinterpret_cast<uintptr_t>(pointer) % alignof(T) == 0;
 }
 
+#if !defined(NDEBUG)
```
Comment (TheMarex) [inline src/contractor/contractor.cpp]: Unrelated change?

Comment (oxidase) [inline src/contractor/contractor.cpp]: I will open another PR to fix current warnings, just missed this one because it is only related to Release builds

Comment (TheMarex) [inline src/contractor/contractor.cpp]: It's fine, I was just wondering where this came from. 👍
